pkg/epinio/detail/applications.vue

```diff
@@ -84,12 +125,96 @@ export default Vue.extend<Data, any, any, any>({
       const matchGithub = str.match('^(https|git)(:\/\/|@)([^\/:]+)[\/:]([^\/:]+)\/(.+)(.git)*$');
 
       return `${ matchGithub?.[4] }/${ matchGithub?.[5] }`;
-    }
+    },
+    async fetchRepoDetails() {
+      const envs = this.value?.envDetails;
+
+      if (envs.appDeployment ) {
+        const { usernameOrOrg, repo } = JSON.parse(envs.appDeployment);
+        const res = await this.$store.dispatch('github/fetchRepoDetails', { username: usernameOrOrg, repo });
+
+        this.gitSource = res;
+
+        const commit = this.value.sourceInfo?.details.filter((ele: { label: string; }) => ele.label === 'Revision')[0]?.value;
+
+        if (commit) {
+          this.gitDeployment.deployedCommit = {
+            short: commit?.slice(0, 7),
+            long:  commit
+          };
+        }
+      }
+
+      await this.fetchCommits();
+    },
+    async fetchCommits() {
+      const envs = this.value?.envDetails;
+
+      if (!envs.appDeployment) {
+        return;
+      }
+
+      const { usernameOrOrg, repo, branch } = JSON.parse(envs.appDeployment);
+
+      this.gitDeployment.commitsArray = await this.$store.dispatch('github/fetchCommits', {
+        username: usernameOrOrg, repo, branch
+      });
+    },
+    formatDate(date: string, from: boolean) {
+      day.extend(relativeTime);
+
+      return from ? day(date).fromNow() : day(date).format('DD MMM YYYY');
```

Comment:
The date should be formatted given the current locale. If we do this elsewhere in the dashboard feel free to ignore